cmd/client-s3.go

```diff
@@ -2313,3 +2315,55 @@ func (c *S3Client) SetVersioning(ctx context.Context, status string) *probe.Erro
 	}
 	return probe.NewError(err)
 }
+
+// GetEncryption - gets bucket encryption info.
+func (c *S3Client) GetEncryption(ctx context.Context) (algorithm, keyID string, err *probe.Error) {
+	bucket, _ := c.url2BucketAndObject()
+	if bucket == "" {
+		return "", "", probe.NewError(BucketNameEmpty{})
+	}
+
+	config, e := c.api.GetBucketEncryption(ctx, bucket)
+	if e != nil {
+		return "", "", probe.NewError(e)
+	}
+	for _, rule := range config.Rules {
+		algorithm = rule.Apply.SSEAlgorithm
+		if rule.Apply.KmsMasterKeyID != "" {
+			keyID = rule.Apply.KmsMasterKeyID
+			break
+		}
+	}
+	return algorithm, keyID, nil
+}
+
+// SetEncryption - Set encryption configuration on a bucket
+func (c *S3Client) SetEncryption(ctx context.Context, encType string, kmsKeyID string) *probe.Error {
+	bucket, _ := c.url2BucketAndObject()
+	if bucket == "" {
+		return probe.NewError(BucketNameEmpty{})
+	}
+	var config *sse.Configuration
+	switch strings.ToLower(encType) {
```

Comment:
Let's error out in case of default (I know it is just a defensive code)

Reply:
done